src/legacy/core_plugins/kibana/public/discover/angular/discover.js

```diff
@@ -189,12 +189,14 @@ function discoverController(
   config,
   kbnUrl,
   localStorage,
-  uiCapabilities
+  uiCapabilities,
+  getAppState,
+  globalState,
 ) {
   const responseHandler = vislibSeriesResponseHandlerProvider().handler;
   const getUnhashableStates = Private(getUnhashableStatesProvider);
+  new FilterStateManager(globalState, getAppState, filterManager);
```

Comment:
I don't think you should initialize multiple instances of `FilterStateManager` per app.
I suspect it might cause events to fire twice.

Reply:
thx, I will look for an alternative

Reply:
I've solved it by calling the FilterStateManagers `destroy` method when the scope is destroyed. functional tests are looking good (1 to go)